jupyter_server/terminal/api_handlers.py

```diff
@@ -21,7 +21,11 @@ def get(self):
     @web.authenticated
     def post(self):
         """POST /terminals creates a new terminal and redirects to it"""
-        name, _ = self.terminal_manager.new_named_terminal()
+        data = self.get_json_body()
+        if data and data.get('cwd'):
```

Comment:
If we're allowing `new_named_terminal` to accept `**kwargs`, is there a particular reason to hardcode `cwd` here? Should we pass the whole JSON body to `new_named_terminal` as kwargs?

Reply:
Makes sense to me